snuba/utils/schemas.py

```diff
@@ -558,3 +557,69 @@ def set_modifiers(self, modifiers: Optional[TModifiers]) -> Enum[TModifiers]:
 
     def get_raw(self) -> Enum[TModifiers]:
         return Enum(self.values)
+
+
+class Tuple(ColumnType[TModifiers]):
+    def __init__(
+        self,
+        types: tuple[ColumnType[TModifiers], ...],
+        modifiers: Optional[TModifiers] = None,
+    ) -> None:
+        super().__init__(modifiers)
+        self.types = types
+
+    def _repr_content(self) -> str:
+        return ", ".join("{}".format(repr(v)) for v in self.types)
+
+    def __eq__(self, other: object) -> bool:
+        return (
+            self.__class__ == other.__class__
+            and self.get_modifiers() == cast(Enum[TModifiers], other).get_modifiers()
+            and self.types == cast(Tuple[TModifiers], other).types
+        )
+
+    def _for_schema_impl(self) -> str:
+        return "Tuple({})".format(", ".join([repr(t) for t in self.types]))
```

Comment:
The repr is not necessarily the same as the schema representation

```suggestion
        return "Tuple({})".format(", ".join([t.for_schema() for t in self.types]))
```